lib/tasks/whitehall_importer.rb

```diff
@@ -120,29 +127,85 @@ def create_edition(document, translation, whitehall_edition, edition_number)
         created_at: whitehall_edition["created_at"],
       )
 
-      Edition.create!(
+      edition = Edition.create!(
         document: document,
         number: edition_number,
         revision_synced: true,
         revision: revision,
-        status: Status.new(
-          state: state(whitehall_edition),
-          revision_at_creation: revision,
-        ),
+        status: initial_status(whitehall_edition, revision),
         current: whitehall_edition["id"] == most_recent_edition["id"],
         live: live?(whitehall_edition),
         created_at: whitehall_edition["created_at"],
         updated_at: whitehall_edition["updated_at"],
         created_by_id: user_ids[first_author["whodunnit"]],
         last_edited_by_id: user_ids[last_author["whodunnit"]],
       )
+
+      set_current_status(whitehall_edition, edition) if set_published_state_first?(whitehall_edition["state"])
```

Comment:
It looks like this is setting the first and last status of an edition, but since an edition can be published/withdrawn multiple times, should we be storing every status change to capture the events between first and last?

Reply:
I think I only need to store the most recent withdrawal details for the edition. I'll change this to get the `last` revision history.

Reply:
Sorry to chime in but an edition can't be published and withdrawn multiple time's in whitehall without creating a new edition each time it's published (unwithdrawn) ? An edition can however be unpublished (published in error or consolidated) and published multiple times on the same edition but not withdrawn to my understanding.